### ISSUE TITLE
interpretation of the srl result

### ISSUE THREAD

**Issue by aleissaf** [biplab-iitb/practNLPTools#2]
Dear,

I have been using the tool for semantic role labelling, but I haven't been able to understand the result of the semantic role labelling. Can someone please guide me.
Text:
RT @WBBMNewsradio: Gov. Rauner Sends National Guard To Southern Illinois In Response To Flooding - Full Story: 
Result
[{'A1': 'National Guard', 'A0': 'Gov. Rauner', 'A2': 'To Southern Illinois', 'AM-PNC': 'In Response To Flooding - Full Story', 'V': 'Sends'}]

Thanks,

**Comment by profversaggi** [comment]
We're using it successfully for one of out projects. I've been able to
gather a good bit of that information from a different source but it might
be useful....let me know if you still need it.

On Mon, Feb 22, 2016 at 3:36 PM, aleissaf notifications@github.com wrote:

> Dear,
> 
> I have been using the tool for semantic role labelling, but I haven't been
> able to understand the result of the semantic role labelling. Can someone
> please guide me.
> Text:
> RT @WBBMNewsradio: Gov. Rauner Sends National Guard To Southern Illinois
> In Response To Flooding - Full Story:
> Result
> [{'A1': 'National Guard', 'A0': 'Gov. Rauner', 'A2': 'To Southern
> Illinois', 'AM-PNC': 'In Response To Flooding - Full Story', 'V': 'Sends'}]
> 
> Thanks,
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/biplab-iitb/practNLPTools/issues/2.

## 

#########################################################
Matthew R. Versaggi,
Artificial Intelligence Engineer, Imagine One, LTD
President & CEO: Versaggi Information Systems, Inc.
Adjunct Professor of eBusiness DePaul University
Email: mailto:matt@versaggi.com, ProfVersaggi@gmail.com
M: 630-292-8422
LinkedIn:  http://www.linkedin.com/in/versaggi
About Me: http://www.matt-versaggi.com/resume/
#########################################################

**Comment by aleissaf** [comment]
Dear,

As there any way that I can extract only the events from the semantic role labelling?

Thanks,

**Comment by profversaggi** [comment]
Yes, we've done specifically that in our project. I have extensive notes on
exactly how that is done. We used SRL and PyKE (inference engine) to
accomplish that. It's a fusion of technologies but it works.

If you like I can forward what I have to you in order to walk you through
it ...

On Fri, Feb 26, 2016 at 5:58 PM, aleissaf notifications@github.com wrote:

> Dear,
> 
> As there any way that I can extract only the events from the semantic role
> labelling?
> 
> Thanks,
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/biplab-iitb/practNLPTools/issues/2#issuecomment-189517361
> .

## 

#########################################################
Matthew R. Versaggi,
Artificial Intelligence Engineer, Imagine One, LTD
President & CEO: Versaggi Information Systems, Inc.
Adjunct Professor of eBusiness DePaul University
Email: mailto:matt@versaggi.com, ProfVersaggi@gmail.com
M: 630-292-8422
LinkedIn:  http://www.linkedin.com/in/versaggi
About Me: http://www.matt-versaggi.com/resume/
#########################################################

**Comment by aleissaf** [comment]
Dear Versaggi,

I would really appreciate it if you can share anything that guides me through the process.

Thank you very much

**Comment by profversaggi** [comment]
I'd be glad to, give me an idea of what it is you are doing and the context
and I'll see if I can cater it better to what you are doing....

On Sun, Feb 28, 2016 at 3:13 PM, aleissaf notifications@github.com wrote:

> Dear Versaggi,
> 
> I would really appreciate it if you can share anything that guides me
> through the process.
> 
> Thank you very much
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/biplab-iitb/practNLPTools/issues/2#issuecomment-189936620
> .

## 

#########################################################
Matthew R. Versaggi,
Artificial Intelligence Engineer, Imagine One, LTD
President & CEO: Versaggi Information Systems, Inc.
Adjunct Professor of eBusiness DePaul University
Email: mailto:matt@versaggi.com, ProfVersaggi@gmail.com
M: 630-292-8422
LinkedIn:  http://www.linkedin.com/in/versaggi
About Me: http://www.matt-versaggi.com/resume/
#########################################################

**Comment by aleissaf** [comment]
Basically, i'm trying to read and analyze twitter news feed. I want to identify from the results only the events after removing the stop words.

**Comment by profversaggi** [comment]
Hi Aleissaf,

I used a different approach. I leveraged an Inference engine to pattern
match only those patterns which would typically correspond to an event.
That worked well for me ...

I've included a process flow diagram which shows how I did it, and can give
you code snippets of that'll help.

In our case we had different requirements, we were picking out only
specific events  in a news feed so we only wanted those events whose verb 
corresponded to a DB of (selected) verbs which we seeded (and would be 
indicative of the kinds of events we were tracking.

In most cases that worked really well for us to capture the universe of
events and then allow us to whittle them down to just what we wanted.

Let me know if this helps and if you need some code to peruse.

-matt

On Mon, Feb 29, 2016 at 7:50 AM, aleissaf notifications@github.com wrote:

> Basically, i'm trying to read and analyze twitter news feed. I want to
> identify from the results only the events after removing the stop words.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/biplab-iitb/practNLPTools/issues/2#issuecomment-190196952
> .

## 

#########################################################
Matthew R. Versaggi,
Artificial Intelligence Engineer, Imagine One, LTD
President & CEO: Versaggi Information Systems, Inc.
Adjunct Professor of eBusiness DePaul University
Email: mailto:matt@versaggi.com, ProfVersaggi@gmail.com
M: 630-292-8422
LinkedIn:  http://www.linkedin.com/in/versaggi
About Me: http://www.matt-versaggi.com/resume/
#########################################################

**Comment by profversaggi** [comment]
I don't think that the process flow diagram came through via the email response, so I added it here...

![srl-pyke-process-flow](https://cloud.githubusercontent.com/assets/1290117/13476594/92d957e2-e096-11e5-9be2-dcc6a8b8fccd.png)

**Comment by subhadeepmaji** [comment]
Please refer to this for a detailed description of Propbank SRL labels. 
https://verbs.colorado.edu/~mpalmer/projects/ace/PBguidelines.pdf

SENNA uses Propbank labeling, other popular choice is FrameNet.
